src/ddmd/traits.d

```diff
@@ -910,6 +911,35 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         auto tup = new TupleExp(e.loc, mods);
         return tup.semantic(sc);
     }
+    if (e.ident == Id.getLinkage)
+    {
+        // get symbol linkage as a string
+        if (dim != 1)
+            return dimError(1);
+
+        auto o = (*e.args)[0];
+        auto s = getDsymbol(o);
+        Declaration d;
+        if (!s || (d = s.isDeclaration()) is null)
+        {
+            e.error("argument to `__traits(getLinkage, %s)` is not a declaration", o.toChars());
+            return new ErrorExp();
+        }
+        string linkage;
+        switch (d.linkage)
+        {
```

Comment:
Maybe ... rather then using strings here we take the pointers to the identifiers out of the string table we already have ?
Id.{D, C, Windows ... }.string